airflow/utils/setup_teardown.py

```diff
@@ -16,42 +16,49 @@
 # under the License.
 from __future__ import annotations
 
-from contextlib import contextmanager
+import contextlib
+import enum
 
 from airflow.exceptions import AirflowException
 
 
+class SetupTeardown(enum.Enum):
+    """Flag to indicate a setup/teardown context."""
+
+    setup = "setup"
+    teardown = "teardown"
```

Comment:
Out of curiosity, while I can (almost) follow *what* this is doing, I'm not sure of the *why* — what is the advantage to this refactor?